### ISSUE TITLE
NET::ERR_CERT_AUTHORITY_INVALID

### ISSUE THREAD

**Issue by zhex900** [coreos/tectonic-installer#3109]
# BUG REPORT

### Tectonic version ([release](https://github.com/coreos/tectonic-installer/releases) or commit hash):
```
latest
```

### Terraform version (`terraform version`):

```
latest
```
### Platform:

```
AWS
```

### What happened?

I have followed the GUI cluster install. After the successful install is completed. The console page cannot be accessed. 
```
Your connection is not private
Attackers might be trying to steal your information from k8.ewhale.co (for example, passwords, messages or credit cards). Learn more
NET::ERR_CERT_AUTHORITY_INVALID
Subject: k8.ewhale.co

Issuer: kube-ca

Expires on: 13 Mar 2021

Current date: 14 Mar 2018

PEM encoded chain:
-----BEGIN CERTIFICATE-----
MIIDuDCCAqCgAwIBAgIQXzgfoWUsTyjNOs97s+LcVDANBgkqhkiG9w0BAQsFADBc
MQkwBwYDVQQGEwAxCTAHBgNVBAgTADEJMAcGA1UEBxMAMQkwBwYDVQQREwAxETAP
BgNVBAoTCGJvb3RrdWJlMQkwBwYDVQQLEwAxEDAOBgNVBAMTB2t1YmUtY2EwHhcN
MTgwMzEzMjMzMTEyWhcNMjEwMzEyMjMzMTEyWjBZMQkwBwYDVQQGEwAxCTAHBgNV
BAgTADEJMAcGA1UEBxMAMQkwBwYDVQQREwAxCTAHBgNVBAoTADEJMAcGA1UECxMA
MRUwEwYDVQQDEwxrOC5ld2hhbGUuY28wggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAw
ggEKAoIBAQCsqyFzlB1RkCSxZNRmVXmj9ousuLrhN+YL337wqMI77VWRtImy4UkH
WgL38k9v8A3DCJQCKong42qUMLWp24r1noCvzl6FBN5KbGmoGfgGdF2Dm8qfoVoW
LIYk6BaB2s8XEiNHLfjLQTuOaS5x8t+Xh6RQvC+8gLEZhoxVmDrUwf9u7JcwNMRK
SweWwuK8G4Lk/v7VrjDDXmDteHZS9J/lyYN0WOpcOHlQPB6tG02iXrDds9Aafta+
lyOqMrmF9A4o+C4aILAG0jw4valg5OL316LIMBWuGn+pRo0e+UvuXb4fQ5IulbbE
uIzpzMD6CJ+7Yz39XRBT8iXNGHuaJ8QNAgMBAAGjeTB3MA4GA1UdDwEB/wQEAwIF
oDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwDAYDVR0TAQH/BAIwADAf
BgNVHSMEGDAWgBQfwP7OE9Klo00Cdt06SzGlg9mdrDAXBgNVHREEEDAOggxrOC5l
d2hhbGUuY28wDQYJKoZIhvcNAQELBQADggEBAL2Q81qv91VnLbtXeHA3erhRPS3K
pE/qIdbi7qJX5q25t5q779soDroDEI1igwNpGwZ1Ufg/RKA1ER0zdol7mJxUoaKq
XUzk6XKcaS1ww3LotYoinCRT1LgOOjCPA+x8CuKPAAefPuUtH5Nt/zLGbYxKsCjm
iiKvGKGNy0qAZEQPFTjRmfthWOq6Sr9IgX6AVjSLVDwu6ulfnRpjNnAxG8DtzfWp
AkP2xRJSKI8Tf33DaoprnHFn10tDnV/OoPKPHPnTjAQI4WHdXQgLF6UHTMy8seQn
RL7lx6HViPfUqi342sh7huAagGEK5Ha5Lqdl0ChtkL9uYS4bTsERHoZm8Cw=
-----END CERTIFICATE-----

 
Automatically send some system information and page content to Google to help detect dangerous apps and sites. Privacy Policy
k8.ewhale.co normally uses encryption to protect your information. When Google Chrome tried to connect to k8.ewhale.co this time, the website sent back unusual and incorrect credentials. This may happen when an attacker is trying to pretend to be k8.ewhale.co, or a Wi-Fi sign-in screen has interrupted the connection. Your information is still secure because Google Chrome stopped the connection before any data was exchanged.

You cannot visit k8.ewhale.co right now because the website uses HSTS. Network errors and attacks are usually temporary, so this page will probably work later.
```

What should I do to fix it?

### References

https://github.com/coreos/tectonic-installer/issues/2189

**Comment by squat** [comment]
@zhex900, the tectonic installer allows you to cobfigure your PKI two ways:
1. it generates a self-signed CA and uses this to sign all certificates, including those used for console; or
2. it allows you to provide certificates that you have generated that were signed by a CA that is valid for the given domain

The issue with the first approach is that your laptop will not trust the CA and by extension the certificates it has signed by default. You need to either trust the individual certificates or add the CA to your OS’s trusted CA store.

In other words, it is expected that your browser earns you about an unknown certificate authority if you have not explicitly trusted the generated CA. If you ignore/bypass the warning, can you proceed to the console? If so, then the installer is working as expected.

**Comment by nreisbeck** [comment]
@squat This is a valid BR.

My PR #2844 was merged to master, but it was apparently never merged to the working branches.

``1.8.7-tectonic.2`` is [still shipping with tls module v1.0.0](https://github.com/coreos/tectonic-installer/blob/1.8.9/config.tf#L33-L35) and I don't see the fix included in the release notes for any other release branches.

From @zhex900's PEM above:
``openssl verify cert``
```
cert: C = , ST = , L = , postalCode = , O = , OU = , CN = k8.ewhale.co
```

``openssl x509 -in cert -text -noout``
```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            5f:38:1f:a1:65:2c:4f:28:cd:3a:cf:7b:b3:e2:dc:54
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: C=, ST=, L=/postalCode=, O=bootkube, OU=, CN=kube-ca
        Validity
            Not Before: Mar 13 23:31:12 2018 GMT
            Not After : Mar 12 23:31:12 2021 GMT
        Subject: C=, ST=, L=/postalCode=, O=, OU=, CN=k8.ewhale.co
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
```
I just tested with ``1.8.7-tectonic.2`` on a test cluster, and the x509 is still invalid per RFC specs and being rejected as unparsable by browsers. 
![image](https://user-images.githubusercontent.com/10735102/37424854-c29d23b0-277e-11e8-9311-4b0b51e11c65.png)

@zhex900 In the meantime until this is merged from master into the version branches, you can simply change ``version = "1.0.0"`` to ``version = "1.0.1"`` in ``config.tf``.

**Comment by zhex900** [comment]
@nreisbeck  Thank you!

**Comment by nreisbeck** [comment]
@zhex900 would you mind re-opening this issue until someone from CoreOS has a chance to respond about when this will be fixed on branch?

**Comment by sym3tri** [comment]
Seems this was fixed in https://github.com/coreos/tectonic-installer/pull/3114
